src/libraries/Builder/Microsoft.Agents.Builder.Dialogs/DialogSet.cs

```diff
@@ -92,11 +92,11 @@ public DialogSet Add(Dialog dialog)
 
             AssertionHelpers.ThrowIfNull(dialog, nameof(dialog));
 
-            if (_dialogs.ContainsKey(dialog.Id))
+            if (_dialogs.TryGetValue(dialog.Id, out Dialog value))
             {
                 // If we are trying to add the same exact instance, it's not a name collision.
                 // No operation required since the instance is already in the dialog set.
-                if (_dialogs[dialog.Id] == dialog)
+                if (value == dialog)
```

Comment:
Can you alter the suggestion there to use a name other than Value (makes my head hurt b/c its a context verb in C# )